cmd/minikube/cmd/config/update.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2022 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/docker/machine/libmachine"
+	"github.com/docker/machine/libmachine/host"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/minikube/pkg/minikube/config"
+	"k8s.io/minikube/pkg/minikube/driver"
+	"k8s.io/minikube/pkg/minikube/exit"
+	"k8s.io/minikube/pkg/minikube/localpath"
+	"k8s.io/minikube/pkg/minikube/machine"
+	"k8s.io/minikube/pkg/minikube/mustload"
+	"k8s.io/minikube/pkg/minikube/out"
+	"k8s.io/minikube/pkg/minikube/reason"
+	"k8s.io/minikube/pkg/minikube/registry"
+	"k8s.io/minikube/pkg/minikube/style"
+)
+
+var configUpdateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Update machine config",
+	Long:  `Update machine config, for e.g. when the minikube home folder is moved to other location.`,
+	Run: func(command *cobra.Command, args []string) {
+		api, err := machine.NewAPIClient()
+		if err != nil {
+			exit.Message(reason.NewAPIClient, "NewAPIClient")
+		}
+
+		cname := ClusterFlagValue()
+		_, cc := mustload.Partial(cname)
+
+		for _, n := range cc.Nodes {
+			h, err := api.Load(config.MachineName(*cc, n))
+			if err != nil {
+				exit.Message(reason.HostConfigLoad, "Error loading existing host.")
+			}
+
+			nHost, err := newHost(api, cc, &n)
+			if err != nil {
+				out.Styled(style.Failure, "Error creating new host")
+				os.Exit(1)
+			}
+
+			fixAuthOptionsPaths(h, nHost)
+
+			err = fixDriverPaths(h, nHost)
+			if err != nil {
+				out.Styled(style.Failure, "Error fixing driver paths")
+				os.Exit(1)
+			}
+
+			if err := api.Save(h); err != nil {
+				exit.Message(reason.HostSaveProfile, "Save host failed")
```

Comment:
The actual error (`err`) is never output, would be good to have it for debugging purposes.

Reply:
Done.